src/components/form/HdInputPhone.vue

```diff
@@ -0,0 +1,368 @@
+<template>
+  <div class="phone-input">
+    <button
+      class="phone-input__selector"
+      aria-haspopup="listbox"
+      aria-label="Choose a country code:"
+      @click="toggleDropdown"
+      :aria-activedescendant="`${selectedCountry.dial_code}, ${t.COUNTRIES[selectedCountry.code]}`"
+    >
+      <span :class="`phone-input__selector__flag flag-icon flag-icon-${selectedCountry.code.toLowerCase()}`"></span>
+      <HdIcon
+        width="24px"
+        height="24px"
+        :src="smallArrowIcon"
+        :class="arrowClassNames"
+      />
+    </button>
+    <ul
+      id="country-codes-list"
+      tabindex="-1"
+      role="listbox"
+      v-show="showDropdown"
+      aria-label="Choose a country code:"
+      class="phone-input__dropdown"
+    >
+      <template>
+        <li
+          v-for="(country, index) in sortedCountriesList"
+          :key="country.code"
+          :id="country.code"
+          class="phone-input__dropdown__option"
+          role="option"
+          :aria-selected="selectedCountry.code === country.code"
+          :ref="country.code"
+        >
+          <button @click="selectCountry(country)" class="option-country">
+            <span :class="`option-country__flag flag-icon flag-icon-${country.code.toLowerCase()}`"></span>
+            <p class="option-country__name-and-code">
+              <span class="option-country__name">{{t.COUNTRIES[country.code]}}</span>
+              <span class="option-country__dial-code">{{country.dial_code}}</span>
+            </p>
+          </button>
+          <hr v-if="index === preferredCountries.length - 1" class="phone-input__dropdown__divider" />
+        </li>
+      </template>
+    </ul>
+    <HdInputFormatter
+      ref="input"
+      type="tel"
+      name="phoneNumber"
+      label="Telefonnumer"
+      v-model="phoneNumber"
+      class="phone-input__field"
+      placeholder="+55 555 555 5555"
+      :custom-rules="[rule]"
+      :formatter="phoneFormatter"
+      @input="handleInputEvent"
+    />
```

Comment:
Why do we use `HdInputFormatter` component even we format our phone number via `phoneFormatter` function (::210)?

Reply:
Oh, the HdInputFormatter needs a function to format the value as we want, that's why it's there

Reply:
Hmm, interesting, I don't know if we need `HdInputFormatter` component in this case, but don't worry, each solution works for me, I will resolve the conversation, thanks for your answer.